cadc-util/src/main/java/ca/nrc/cadc/auth/BearerTokenPrincipal.java

```diff
@@ -75,12 +75,17 @@
 import java.security.Principal;
 
 /**
- * Represents a bearer token
+ * Represents a bearer token.
+ * 
+ * @deprecated Being replaced with more general token support in AuthorizationToken
+ *     and AuthorizationTokenPrincipal
  */
+@Deprecated 
```

Comment:
If in addition to deprecating BearerTokenPrincipal you made it a subclass of AuthorizationToken you wouldn't need to add two principals for compatibility. You could just add this thing for awhile and then switch to AuthorizationToken when this class gets deleted. That's a common and useful pattern when a class gets replaced by something else.